crypto.go

```diff
@@ -46,7 +46,7 @@ func publicKeyFromProto(pbKey *proto.Key) (PublicKey, error) {
 		return NewKeyList().AddAll(keys), nil
 
 	default:
-		return nil, fmt.Errorf("key type not implemented: %v", key)
+		return nil, newErrBadKeyf("key type not implemented: %v", key)
```

Comment:
Again Java SDK does this, but I'll wait for a response from @mehcode or @abonander on their thoughts.

I disagree though, bad key error is used in multiple contexts (this may be its own problem though). Having a message is useful in some cases, especially in the when it is an unsupported key type or because the keyfile couldn't be parsed properly.

Another idea would be to have more specific errors in those situations (ErrUnsupportedKey and ErrMalformedKeystore or something), which would improve the error checking more in go, but be problematic in other situations but would allow easier obfuscation of key specific errors.

Reply:
The consensus for this is the more information you have about the error the better, but what is important is the error type is called `BadKey`.